client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/Connection.spec.ts

```diff
@@ -191,6 +191,87 @@ describe("Connection", () => {
         done();
     });
 
+    describe("negotiate request", () =>
+    {
+        let negotiateUrl: string;
+        let connection: HttpConnection;
+        let options: IHttpConnectionOptions = {
+            httpClient: <IHttpClient>{
+                post(url: string): Promise<string> {
+                    negotiateUrl = url;
+                    connection.stop();
+                    return Promise.resolve("{}");
+                },
+                get(url: string): Promise<string> {
+                    connection.stop();
+                    return Promise.resolve("");
+                }
+            },
+            logging: null
+        } as IHttpConnectionOptions;
+
+        it("preserves user query with / before query", async done =>
```

Comment:
make these tests theory-like? They are all the same except the input and output...